### ISSUE TITLE
vmbackup: snapshots not deleted from db

### ISSUE THREAD

**Issue by k-stz** [VictoriaMetrics/VictoriaMetrics#3858]
### Describe the bug

Hello team,

Previously we experienced sporadic hanging of snapshot creation when calling the `snapshot/create` api (see my comment on issue #3551).
After updating the Image from `v1.79.5` to latest `v1.87.0` we get problems when creating backups with `vmbackup:1.87.0`: snapshots get created but not deleted at sporadic times from the database `/snapshots` dir.

## Expected behavior
When we're calling from container `vmbackup:v1.87.0`:
```shell
/vmbackup-prod -storageDataPath=/vm-db-path
               -snapshot.createURL=http://vmsingle:<port>/snapshot/create
               -dst=fs:///vm-data-backup/latest`
```
It should create the snapshot in (1) `/vm-db-path/snapshots`, (2) copy it to `fs:///vm-data-backup/latest` and delete the snapshot from (1).

## Problem
It sporadically doesn't delete the snapshot, the victoriametrics database throws an error in its logs (see below "Error caught in the wild" for full example):
```log
cannot create snapshot: cannot create curr indexDB snapshot: cannot create hard links from "/vm-db-path/indexdb/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233" to "/vm-db-path/indexdb/snapshots/20230222100014-1742279E7D4385A9/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233": cannot open srcDir: open /vm-db-path/indexdb/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233: no such file or directory
```
This happens roughly twice a day. Since the jobs runs hourly this happens 2 out of 24 times.

# Error caught in the wild
For example an error occured today (22.02.2023) at 10:00 o'clock UTC:
- Images used:
    - VictoriaMetrics from Dockerhub: `victoriametrics/victoria-metrics:v1.87.0`
    - VMbackup from Dockerhub: `/victoriametrics/vmbackup:v1.87.0`
    - (Since v1.87.1 Release Notes don't mention anything about snapshots or backups, I think this bug report is still valid.)
- Problem: 
    - Snapshot not deleted: `20230222100014-1742279E7D4385A9` it still exists in in: `/vm-db-path/snapshot/20230222100014-1742279E7D4385A9`
    - In fact it exists in multiple locations in db:
        ```shell
        /vm-db-path $ find . -name "20230222100014-1742279E7D4385A9" -type d

        ./data/big/snapshots/20230222100014-1742279E7D4385A9
        ./data/small/snapshots/20230222100014-1742279E7D4385A9
        ./indexdb/snapshots/20230222100014-1742279E7D4385A9
        ./snapshots/20230222100014-1742279E7D4385A9
        ```
- Expected behaviour:
    - Via vmbackup image: After snapshot creation (api `snapshot/create`) snapshot gets created, then copied over to backup destination
    - then created snapshot in Victoriametrics Stoarge gets deleted
- Maybe related OpenShift Events:
    - 10:00 UTC: 
        - `"failed to ensurePod my-namespac/vmbackup-hourly-some-hash since it is not yet scheduled"`
        - Maybe this is related? Since the vmbackup-cronjob-pod mounts two volumes one with the vm-database and one with the vm-backups
- VictoriaMetrics Logs at this time:
```log
1742279E7D4385A9/2023_02" and "/vm-db-path/data/big/snapshots/20230222100014-1742279E7D4385A9/2023_02" in 2.707 seconds
2023-02-22T10:00:17.007Z	info	VictoriaMetrics/lib/storage/partition.go:1947	creating partition snapshot of "/vm-db-path/data/small/2023_01" and "/vm-db-path/data/big/2023_01"...
2023-02-22T10:00:17.209Z	info	VictoriaMetrics/lib/storage/partition.go:1965	created partition snapshot of "/vm-db-path/data/small/2023_01" and "/vm-db-path/data/big/2023_01" at "/vm-db-path/data/small/snapshots/20230222100014-1742279E7D4385A9/2023_01" and "/vm-db-path/data/big/snapshots/20230222100014-1742279E7D4385A9/2023_01" in 0.201 seconds
2023-02-22T10:00:17.210Z	info	VictoriaMetrics/lib/storage/table.go:174	created table snapshot for "/vm-db-path/data" at ("/vm-db-path/data/small/snapshots/20230222100014-1742279E7D4385A9", "/vm-db-path/data/big/snapshots/20230222100014-1742279E7D4385A9") in 2.916 seconds
2023-02-22T10:00:17.211Z	info	VictoriaMetrics/lib/mergeset/table.go:1496	creating Table snapshot of "/vm-db-path/indexdb/1742279E7D44822B"...
2023-02-22T10:00:19.229Z	error	VictoriaMetrics/app/vmstorage/main.go:902	cannot create snapshot: cannot create curr indexDB snapshot: cannot create hard links from "/vm-db-path/indexdb/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233" to "/vm-db-path/indexdb/snapshots/20230222100014-1742279E7D4385A9/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233": cannot open srcDir: open /vm-db-path/indexdb/1742279E7D44822B/1645_8_1744320EB34F176F.must-remove.1675945572994494233: no such file or directory
```
=> Here the Error "no such file or directory" hints to for me to a bug. The vmbackup job runs every hour and only fails roughly two times a day at completely random hours. I think that when this occured with the `v1.79.5` image this caused the api to hang indefinetly as described in issue #3551

- Vmbackup Logs:
    Via cronjob we run vmbackup, it reports deleting the snapshot.After the logs show that the backup was created successfully it shows:
```log
15529133054 bytes in 2.261 seconds; deleted 50669498 bytes; server-side copied 0 bytes; uploaded 81161490 bytes
2023-02-22T10:00:24.646Z	info	VictoriaMetrics/app/vmbackup/main.go:124	gracefully shutting down http server for metrics at ":8420"
2023-02-22T10:00:24.646Z	info	VictoriaMetrics/app/vmbackup/main.go:128	successfully shut down http server for metrics in 0.000 seconds
2023-02-22T10:00:24.646Z	info	VictoriaMetrics/lib/snapshot/snapshot.go:63	Deleting snapshot 20230222100020-1742279E7D4385AA
2023-02-22T10:00:26.294Z	info	VictoriaMetrics/lib/snapshot/snapshot.go:90	Snapshot 20230222100020-1742279E7D4385AA deleted
```
Note that the deleted snapshots in the vmbackup logs `20230222100020-1742279E7D4385AA` is not the one that still remains in the `/vm-db-path/snapshot` dir

### To Reproduce

Victoriametrics gets deployed via the victoriametrics operator `CR/vmsingle`.
The backup runs as a k8s/cronjob every hour. With the command:
```shell
/vmbackup-prod -storageDataPath=/vm-db-path
               -snapshot.createURL=http://vmsingle:<port>/snapshot/create
               -dst=fs:///vm-data-backup/latest`
```
It has the victoriametrics PVC mounted to: `/vm-db-path` and a backup volume mounted to `/vm-data-backup`

### Version

victoria-metrics-20230201-220651-tags-v1.87.0-0-gfe736c538

### Used command-line flags

```yaml
image: '<docker-io-mirror>/victoriametrics/victoria-metrics:v1.87.0'
      args:
        - '-envflag.enable=true'
        - '-httpListenAddr=:<port>'
        - '-loggerLevel=INFO'
        - '-retentionPeriod=6w'
        - '-storageDataPath=/vm-db-path'
```

**Comment by valyala** [comment]
@k-stz , thanks for filing the bugreport! It should be fixed in the commit d8eaa511b06cbc41e79e4c8298e00772319f3c06 . This commit will be included in the next release. In the mean time you can build VictoriaMetrics from this commit according to [these docs](https://docs.victoriametrics.com/#how-to-build-from-sources) and verify whether the issue is fixed now.

**Comment by valyala** [comment]
The bugfix has been included in [v1.87.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed then.

@k-stz , feel free re-opening this issue or creating new one if `v1.87.2` still has issues related to this one.

**Comment by k-stz** [comment]
Awesome, that was quick, much appreciated! Will be testing this very soon!

**Comment by valyala** [comment]
FYI, the bugfix has been included also into [v1.88.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Comment by valyala** [comment]
FYI, VictoriaMetrics could leave some partially created directories of the snapshot on failure to create the snapshot. This has been fixed in [v1.88.1](https://docs.victoriametrics.com/CHANGELOG.html) - now VictoriaMetrics deletes all the partially created directories on unsuccessful attempt to create the snapshot.

**Comment by k-stz** [comment]
FYI:
I'm glad to announce that we've tested the new version over the last two days that now everything works as expected 😄
